### PR TITLE
Add ReduceOverNeighbor

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

File: src/dawn/Optimizer/AccessComputation.cpp [hunk 1]
```diff
@@ -256,6 +256,7 @@ class AccessMapper : public iir::ASTVisitor {
     // the block statements as the if/then/else block of the if-statement has been collapsed into
     // one single
     // vector of block statements
+    appendNewAccesses();
```
Comment (havogt) [inline src/dawn/Optimizer/AccessComputation.cpp]: This breaks gtclang.

Comment (lukasm91) [reply src/dawn/Optimizer/AccessComputation.cpp]: I am wondering why gtclang was green. I pushed a fix (I hope), but I admit that I don't fully understand how if statements and block statements interact. But block statements are broken in gtclang anyway, so I only fix them for iir as far as they compile and run now.

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch pipeline

**Comment by havogt** [comment]
launch pipeline

**Comment by lukasm91** [comment]
launch pipeline

**Pull request merged**